rust.ungram

```diff
@@ -488,7 +488,7 @@ MatchArm =
   Attr* Pat guard:MatchGuard? '=>' Expr ','?
 
 MatchGuard =
-  'if' Expr
+  'if' ('let' Pat '=')? Expr
```

Comment:
I wonder if we should define MatchGuard in terms of `Condition` now since its basically the same, that is:
```
MatchGuard =
  'if' Condition
```

Reply:
Yeah that could work. One thing I noticed is that the *parser* for `condition` couldn't be reused since it forbids struct literals.